### PR TITLE
32 test cases for apis

### PR DESCRIPTION
**Pull request by rsqwerty**

### CODE REVIEW

File: newscout_web/api/v1/tests.py [hunk 1]
```diff
@@ -1,6 +1,478 @@
...
+User = get_user_model()
+
+client = APIClient()
+user = User.objects.get(username="rushikesh@fafadiatech.com")
+# token = Token.objects.get(user=user)
+# client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+client.force_authenticate(user=user)
+
+
+class NewscoutApiTest(APITestCase):
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add doc string for all methods and class

File: newscout_web/api/v1/tests.py [hunk 2]
```diff
@@ -1,6 +1,478 @@
...
+        data = {
+            "categories": "125",
+            "articles": {},
+        }
+        res = client.post(url, data, format="json")
+        res_status = res.status_code
+        self.assertEqual(res_status, 200)
+
+    def test_get_all_category(self):
+        res_count = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: get all categories objects in def setup and assert all categories

File: newscout_web/api/v1/tests.py [hunk 3]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/categories/"
+        client.put(url, {"id": req_id, "name": req_name}, format="json")
+        res = client.get(url)
+        for x in res.data["body"]["categories"]:
+            if x["id"] == req_id:
+                res_name = x["name"]
+        self.assertEqual(res_name, req_name)
+
+    def test_get_liked_articles(self):
+        res_count = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: implement def setup and create articlelike object and assert same object

File: newscout_web/api/v1/tests.py [hunk 4]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/articles/like-news-list/"
+        res = client.get(url, format="json")
+        res_count = len(res.data["body"]["results"])
+        self.assertGreater(res_count, 0)
+
+    def test_get_fetch_bookmarks(self):
+        res_id = 0
+        req_id = Article.objects.get(id=998398).id
+        url_post_bookmark = "/api/v1/articles/bookmark/"
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: create bookmarkarticle object

File: newscout_web/api/v1/tests.py [hunk 5]
```diff
@@ -1,6 +1,478 @@
...
+        res_msg = ""
+        success_msg = "Successfully saved tags"
+        url = "/api/v1/tags/save/"
+        res = client.post(url, {"tags": "India"}, format="json")
+        res_msg = res.data["body"]["Msg"]
+        self.assertEqual(res_msg, success_msg)
+
+    def test_get_all_sources(self):
+        res_count = 0
+        url = "/api/v1/source/"
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add sources in def setup

File: newscout_web/api/v1/tests.py [hunk 6]
```diff
@@ -1,6 +1,478 @@
...
+        res = client.get(url, format="json")
+        res_count = len(res.data["body"]["results"])
+        self.assertGreater(res_count, 0)
+
+    def test_post_article_bookmark(self):
+        res_msg = ""
+        success_msg = "Article bookmarked successfully"
+        req_id = Article.objects.get(id=998398).id
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: get single article object in def setup. Article.objects.last()

File: newscout_web/api/v1/tests.py [hunk 7]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/articles/bookmark/"
+        res = client.post(url, {"article_id": req_id, "user": user.id})
+        res_msg = res.data["body"]["Msg"]
+        self.assertEqual(res_msg, success_msg)
+
+    def test_get_article_details(self):
+        res_id = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add more assert statements for title, category,source and cover_image

File: newscout_web/api/v1/tests.py [hunk 8]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/login/"
+        data = {
+            "device_name": "A",
+            "device_id": "A",
+            "email": "rushikesh@fafadiatech.com",
+            "password": "test123",
+        }
+        res = client.post(url, data, format="json")
+        res_status = res.status_code
+        self.assertEqual(res_status, 200)
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add more assert statements for res data

File: newscout_web/api/v1/tests.py [hunk 9]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/forgot-password/"
+        data = {
+            "email": "rushikesh@fafadiatech.com",
+        }
+        res = client.post(url, data)
+        res_msg = res.data["body"]["Msg"]
+        self.assertEqual(res_msg, success_msg)
+
+    def test_post_change_password(self):
+        res_msg = ""
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add assert statement for status code

File: newscout_web/api/v1/tests.py [hunk 10]
```diff
@@ -1,6 +1,478 @@
...
+        data = {
+            "device_id": "AndroidNX00G",
+            "device_name": "AndroidNougat",
+        }
+        res = client.post(url, data)
+        res_msg = res.data["body"]["Msg"]
+        self.assertEqual(res_msg, success_msg)
+
+    def test_get_article_search(self):
+        res_count = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: assert domain name in article response

File: newscout_web/api/v1/tests.py [hunk 11]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/article/search/?domain=newscout"
+        res = client.get(url)
+        res_count = len(res.data["body"]["results"])
+        self.assertGreater(res_count, 0)
+
+    def test_get_all_menus(self):
+        res_count = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: get menu object in def setup

File: newscout_web/api/v1/tests.py [hunk 12]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/api/v1/menus/?domain=newscout"
+        res = client.get(url)
+        for x in res.data["body"]["results"]:
+            res_count += len(x["heading"])
+        self.assertGreater(res_count, 0)
+
+    def test_post_new_article(self):
+        res_title_name = ""
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add asserts for all fields

File: newscout_web/api/v1/tests.py [hunk 13]
```diff
@@ -1,6 +1,478 @@
...
+            "published_on": "2019-11-06 12:15:34.870987",
+            "spam": "True",
+            "domain": "1",
+        }
+        res = client.post(url, data, format="json")
+        res_title_name = res.data["body"]["title"]
+        self.assertEqual(res_title_name, req_title_name)
+
+    def test_put_article_update(self):
+        res_title_name = ""
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add asserts for all fields

File: newscout_web/api/v1/tests.py [hunk 14]
```diff
@@ -1,6 +1,478 @@
...
+            "spam": "True",
+            "domain": "1",
+        }
+        client.put(url, data, format="json")
+        url_get_article_details = "/api/v1/articles/" + str(req_id) + "/"
+        res = client.get(url_get_article_details, format="json")
+        res_title_name = res.data["body"]["article"]["title"]
+        self.assertEqual(res_title_name, req_title_name)
+
+    def test_get_daily_digest(self):
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: add "device_id" in url query params. /api/v1/daily-digest/?device_id=something

File: newscout_web/api/v1/tests.py [hunk 15]
```diff
@@ -1,6 +1,478 @@
...
+
+    def test_get_campaign_categories(self):
+        res_category_count = 0
+        url = "/ads/campaigns/categories/"
+        res = client.get(url)
+        res_category_count = len(res.data["body"]["categories"])
+        self.assertGreater(res_category_count, 0)
+
+    def test_post_campaign_creation(self):
+        res_campaign_name = ""
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: assert all fields

File: newscout_web/api/v1/tests.py [hunk 16]
```diff
@@ -1,6 +1,478 @@
...
+            "end_date": "2019-11-06 12:15:34.870987",
+        }
+        client.post(url, data, format="json")
+        res = client.get(url)
+        if res.data["body"][0]["name"] == req_campaign_name:
+            res_campaign_name = res.data["body"][0]["name"]
+        self.assertEqual(res_campaign_name, req_campaign_name)
+
+    def test_put_campaign_update(self):
+        res_change_name = ""
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: assert all fields

File: newscout_web/api/v1/tests.py [hunk 17]
```diff
@@ -1,6 +1,478 @@
...
+        res_types_count = 0
+        url = "/ads/grouptypes/"
+        res = client.get(url)
+        res_groups_count = len(res.data["body"]["groups"])
+        res_types_count = len(res.data["body"]["types"])
+        self.assertGreater(res_groups_count, 0)
+        self.assertGreater(res_types_count, 0)
+
+    def test_get_all_ads(self):
+        res_status = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: check response and add assert

File: newscout_web/api/v1/tests.py [hunk 18]
```diff
@@ -1,6 +1,478 @@
...
+        url = "/ads/schedules/"
+        res = client.get(url)
+        res_status = res.status_code
+        self.assertEqual(res_status, 200)
+
+    def test_get_advertisement(self):
+        res_count = 0
```
Comment (yspanchal) [inline newscout_web/api/v1/tests.py]: check response

**Comment by sidharthshah** [comment]
fixes #32 and #45

**Pull request merged**